src/app/modules/shared_models/wallet_account_item.nim

```diff
@@ -0,0 +1,88 @@
+import strformat
+
+type
+  WalletAccountItem* = ref object of RootObj
+    name: string
+    address: string
+    color: string
+    emoji: string
+    walletType: string
+    path: string
+    keyUid: string
+
+proc setup*(self: WalletAccountItem,
+  name: string = "",
+  address: string = "",
+  color: string = "",
+  emoji: string = "",
+  walletType: string = "",
+  path: string = "",
+  keyUid: string = ""
+  ) =
+    self.name = name
+    self.address = address
+    self.color = color
+    self.emoji = emoji
+    self.walletType = walletType
+    self.path = path
+    self.keyUid = keyUid
+
+proc initWalletAccountItem*(
+  name: string = "",
+  address: string = "",
+  color: string = "",
+  emoji: string = "",
+  walletType: string = "",
+  path: string = "",
+  keyUid: string = ""
+  ): WalletAccountItem =
+  result = WalletAccountItem()
+  result.setup(name,
+    address,
+    color,
+    emoji,
+    walletType,
+    path,
+    keyUid)
+  
+
+proc `$`*(self: WalletAccountItem): string =
+  result = fmt"""WalletAccountItem(
+    name: {self.name},
+    address: {self.address},
+    color: {self.color},
+    emoji: {self.emoji},
+    walletType: {self.walletType},
+    path: {self.path},
+    keyUid: {self.keyUid},
+    ]"""
+
+proc name*(self: WalletAccountItem): string {.inline.} =
+  return self.name
+
+proc `name=`*(self: WalletAccountItem, value: string) {.inline.} =
+  self.name = value
```

Comment:
why not just declaring the attribute with `*` this has exact same effect to make getter and setter public without writing the code

Reply:
Actually we could, though not sure if that will be reflected in inherited classes when you access this prop from the inherited class instance (I guess it should be working), but anyhow I opted in for this approach (personal preference).
https://nim-lang.org/docs/manual.html#procedures-properties